Edit containerd snapshot size calculation

kiashok opened the pull request:
**What type of PR is this?**


/kind bug

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
du.exe which was used to calculate the snapshot folder sizes was not accurate. Changing that to include a script to do the calculation.
The changes have been tested with the modified collect-windows-logs.ps1 script on an AKS windows node.

Fixes #

**Special notes for your reviewer**:


**Release note**:
```
none
```

kiashok commented:
cc @AbelHu @ShiqianTao @junjiezhang1997

AbelHu commented on staging/cse/windows/debug/collect-windows-logs.ps1:
```diff
@@ -226,12 +226,22 @@ else {
-    C:\aks-tools\DU\du.exe /accepteula
-    C:\aks-tools\DU\du.exe -l 1 C:\ProgramData\containerd\root\io.containerd.snapshotter.v1.windows\snapshots\ > "$ENV:TEMP\$timeStamp-du-snapshot-folder-size.txt"
-    $paths += "$ENV:TEMP\$timeStamp-du-snapshot-folder-size.txt"
+  Write-Host "Collecting actual size of snapshot (without sparse file sizes included)"
```
@junjiezhang1997 can we set `-enableSnapshotSize` in one of AgentBaker Windows E2E to cover these lines?

kiashok replied:
@junjiezhang1997 any thoughts?

AbelHu replied:
@kiashok how long does this step take? It may timeout in test with run-command if it takes too long.

kiashok replied:
Hi @AbelHu , it could take long depening on number and size of the snapshots. That is why I have added this command under the "enableAll" or "enableSnapshotSize" flag so that it is not collected by default. Are there any concerns with this?

AbelHu commented on staging/cse/windows/debug/collect-windows-logs.ps1:
```diff
@@ -226,12 +226,22 @@ else {
+  		$totalSize = $totalSize + $_.Length
+  	}
+  	$output = "Sum of " + $i.FullName + " is " + ($totalSize/1MB) + "MB"
+    Add-Content -Path $snapshotSizesResultFilePath -Value $output
```
indent?

kiashok replied:
Hmm on vscode this code chunk seems to be fine. Its only showing up like this in the PR on github:
![image](https://github.com/Azure/AgentBaker/assets/99994218/6905964b-cb98-4c0c-af7a-3e09a0f00f9a)

Let me rebase and force push and see if it changes at all.

kiashok commented:
@AbelHu @ShiqianTao  could you please take a look? Thanks!

msscotb commented on staging/cse/windows/debug/collect-windows-logs.ps1:
```diff
@@ -226,12 +226,21 @@ else {
+  		$totalSize = $totalSize + $_.Length
+  	}
+  	$output = "Sum of " + $i.FullName + " is " + ($totalSize/1MB) + "MB"
+    Add-Content -Path $snapshotSizesResultFilePath -Value $output
```
This should print the total size of the entire directory in addition to the size of each snapshot.

kiashok commented:
@AbelHu  @ShiqianTao have addressed review comments and tested on AKS cluster before pushing new changes:

kiashok commented:
@AbelHu  @ShiqianTao  have addressed review comments and tested on AKS cluster before pushing new changes. The snapshot size calculation will kick in only when ".\collect-windows-logs.ps1 --enableAll true" is specified.
output tested on AKS cluster is as follows:

Sum of C:\ProgramData\containerd\root\io.containerd.snapshotter.v1.windows\snapshots\1 is 5637.39028072357MB
Sum of C:\ProgramData\containerd\root\io.containerd.snapshotter.v1.windows\snapshots\10 is 0.0393075942993164MB
Sum of C:\ProgramData\containerd\root\io.containerd.snapshotter.v1.windows\snapshots\100 is 0.0413846969604492MB
Sum of C:\ProgramData\containerd\root\io.containerd.snapshotter.v1.windows\snapshots\101 is 0.0397558212280273MB
Sum of C:\ProgramData\containerd\root\io.containerd.snapshotter.v1.windows\snapshots\102 is 0.0401782989501953MB
Sum of C:\ProgramData\containerd\root\io.containerd.snapshotter.v1.windows\snapshots\103 is 0.0391712188720703MB
Sum of C:\ProgramData\containerd\root\io.containerd.snapshotter.v1.windows\snapshots\104 is 0.0393409729003906MB
Sum of C:\ProgramData\containerd\root\io.containerd.snapshotter.v1.windows\snapshots\105 is 0.0427627563476563MB
Sum of C:\ProgramData\containerd\root\io.containerd.snapshotter.v1.windows\snapshots\106 is 0.0402727127075195MB
Sum of C:\ProgramData\containerd\root\io.containerd.snapshotter.v1.windows\snapshots\107 is 0.0409450531005859MB
Sum of C:\ProgramData\containerd\root\io.containerd.snapshotter.v1.windows\snapshots\108 is 0.039393424987793MB
Sum of C:\ProgramData\containerd\root\io.containerd.snapshotter.v1.windows\snapshots\109 is 0.0392732620239258MB
Sum of C:\ProgramData\containerd\root\io.containerd.snapshotter.v1.windows\snapshots\11 is 0.0390625MB
Sum of C:\ProgramData\containerd\root\io.containerd.snapshotter.v1.windows\snapshots\110 is 0.0393953323364258MB
Sum of C:\ProgramData\containerd\root\io.containerd.snapshotter.v1.windows\snapshots\111 is 0.0456743240356445MB
Sum of C:\ProgramData\containerd\root\io.containerd.snapshotter.v1.windows\snapshots\112 is 0.0529680252075195MB
Sum of C:\ProgramData\containerd\root\io.containerd.snapshotter.v1.windows\snapshots\113 is 0.653576850891113MB
Sum of C:\ProgramData\containerd\root\io.containerd.snapshotter.v1.windows\snapshots\114 is 0.0390625MB
Sum of C:\ProgramData\containerd\root\io.containerd.snapshotter.v1.windows\snapshots\115 is 0.0390625MB
Sum of C:\ProgramData\containerd\root\io.containerd.snapshotter.v1.windows\snapshots\116 is 0.0390625MB
Sum of C:\ProgramData\containerd\root\io.containerd.snapshotter.v1.windows\snapshots\117 is 0.0390625MB
Sum of C:\ProgramData\containerd\root\io.containerd.snapshotter.v1.windows\snapshots\118 is 0.0390625MB
Sum of C:\ProgramData\containerd\root\io.containerd.snapshotter.v1.windows\snapshots\119 is 0.0390625MB
Sum of C:\ProgramData\containerd\root\io.containerd.snapshotter.v1.windows\snapshots\12 is 0.986061096191406MB
Sum of C:\ProgramData\containerd\root\io.containerd.snapshotter.v1.windows\snapshots\120 is 0.0390625MB
Sum of C:\ProgramData\containerd\root\io.containerd.snapshotter.v1.windows\snapshots\121 is 8.56066226959229MB
Sum of C:\ProgramData\containerd\root\io.containerd.snapshotter.v1.windows\snapshots\122 is 0.0390625MB
Sum of C:\ProgramData\containerd\root\io.containerd.snapshotter.v1.windows\snapshots\123 is 1431.82479667664MB
Sum of C:\ProgramData\containerd\root\io.containerd.snapshotter.v1.windows\snapshots\124 is 3792.45802402496MB
Sum of C:\ProgramData\containerd\root\io.containerd.snapshotter.v1.windows\snapshots\125 is 0.000329017639160156MB
Sum of C:\ProgramData\containerd\root\io.containerd.snapshotter.v1.windows\snapshots\126 is 0.000946044921875MB
Sum of C:\ProgramData\containerd\root\io.containerd.snapshotter.v1.windows\snapshots\127 is 61.7109375MB
Sum of C:\ProgramData\containerd\root\io.containerd.snapshotter.v1.windows\snapshots\128 is 0.000329017639160156MB
Sum of C:\ProgramData\containerd\root\io.containerd.snapshotter.v1.windows\snapshots\129 is 0.000946044921875MB
Sum of C:\ProgramData\containerd\root\io.containerd.snapshotter.v1.windows\snapshots\13 is 0.978744506835938MB
Sum of C:\ProgramData\containerd\root\io.containerd.snapshotter.v1.windows\snapshots\130 is 64.62939453125MB
Sum of C:\ProgramData\containerd\root\io.containerd.snapshotter.v1.windows\snapshots\131 is 62.18798828125MB
Sum of C:\ProgramData\containerd\root\io.containerd.snapshotter.v1.windows\snapshots\132 is 61.25830078125MB
Sum of C:\ProgramData\containerd\root\io.containerd.snapshotter.v1.windows\snapshots\133 is 4MB
Sum of C:\ProgramData\containerd\root\io.containerd.snapshotter.v1.windows\snapshots\134 is 4MB
Sum of C:\ProgramData\containerd\root\io.containerd.snapshotter.v1.windows\snapshots\135 is 4MB
Sum of C:\ProgramData\containerd\root\io.containerd.snapshotter.v1.windows\snapshots\136 is 36MB
Sum of C:\ProgramData\containerd\root\io.containerd.snapshotter.v1.windows\snapshots\137 is 11MB
Sum of C:\ProgramData\containerd\root\io.containerd.snapshotter.v1.windows\snapshots\138 is 36MB
Sum of C:\ProgramData\containerd\root\io.containerd.snapshotter.v1.windows\snapshots\139 is 66.5254440307617MB
Sum of C:\ProgramData\containerd\root\io.containerd.snapshotter.v1.windows\snapshots\14 is 0.0390625MB
Sum of C:\ProgramData\containerd\root\io.containerd.snapshotter.v1.windows\snapshots\140 is 11MB
Sum of C:\ProgramData\containerd\root\io.containerd.snapshotter.v1.windows\snapshots\141 is 19.4760360717773MB
Sum of C:\ProgramData\containerd\root\io.containerd.snapshotter.v1.windows\snapshots\142 is 0.107711791992188MB
Sum of C:\ProgramData\containerd\root\io.containerd.snapshotter.v1.windows\snapshots\143 is 68MB
Sum of C:\ProgramData\containerd\root\io.containerd.snapshotter.v1.windows\snapshots\144 is 36MB
Sum of C:\ProgramData\containerd\root\io.containerd.snapshotter.v1.windows\snapshots\145 is 36MB
Sum of C:\ProgramData\containerd\root\io.containerd.snapshotter.v1.windows\snapshots\146 is 36MB
Sum of C:\ProgramData\containerd\root\io.containerd.snapshotter.v1.windows\snapshots\147 is 36MB
Sum of C:\ProgramData\containerd\root\io.containerd.snapshotter.v1.windows\snapshots\148 is 4MB
Sum of C:\ProgramData\containerd\root\io.containerd.snapshotter.v1.windows\snapshots\149 is 36MB
Sum of C:\ProgramData\containerd\root\io.containerd.snapshotter.v1.windows\snapshots\15 is 0.0473098754882813MB
Sum of C:\ProgramData\containerd\root\io.containerd.snapshotter.v1.windows\snapshots\16 is 2.70160675048828MB
Sum of C:\ProgramData\containerd\root\io.containerd.snapshotter.v1.windows\snapshots\17 is 102.842220306396MB
Sum of C:\ProgramData\containerd\root\io.containerd.snapshotter.v1.windows\snapshots\18 is 0.0390625MB
Sum of C:\ProgramData\containerd\root\io.containerd.snapshotter.v1.windows\snapshots\19 is 0.0390625MB
Sum of C:\ProgramData\containerd\root\io.containerd.snapshotter.v1.windows\snapshots\2 is 4405.04434490204MB
Sum of C:\ProgramData\containerd\root\io.containerd.snapshotter.v1.windows\snapshots\20 is 540.772309303284MB
Sum of C:\ProgramData\containerd\root\io.containerd.snapshotter.v1.windows\snapshots\21 is 16.2714309692383MB
Sum of C:\ProgramData\containerd\root\io.containerd.snapshotter.v1.windows\snapshots\22 is 0.107711791992188MB
Sum of C:\ProgramData\containerd\root\io.containerd.snapshotter.v1.windows\snapshots\23 is 531.560806274414MB
Sum of C:\ProgramData\containerd\root\io.containerd.snapshotter.v1.windows\snapshots\24 is 16.6812133789063MB
Sum of C:\ProgramData\containerd\root\io.containerd.snapshotter.v1.windows\snapshots\25 is 0.107711791992188MB
Sum of C:\ProgramData\containerd\root\io.containerd.snapshotter.v1.windows\snapshots\26 is 532.08606338501MB
Sum of C:\ProgramData\containerd\root\io.containerd.snapshotter.v1.windows\snapshots\27 is 24.0786590576172MB
Sum of C:\ProgramData\containerd\root\io.containerd.snapshotter.v1.windows\snapshots\28 is 0.107711791992188MB
Sum of C:\ProgramData\containerd\root\io.containerd.snapshotter.v1.windows\snapshots\29 is 18.1864242553711MB
Sum of C:\ProgramData\containerd\root\io.containerd.snapshotter.v1.windows\snapshots\3 is 532.739915847778MB
Sum of C:\ProgramData\containerd\root\io.containerd.snapshotter.v1.windows\snapshots\30 is 0.107711791992188MB
Sum of C:\ProgramData\containerd\root\io.containerd.snapshotter.v1.windows\snapshots\31 is 531.323875427246MB
Sum of C:\ProgramData\containerd\root\io.containerd.snapshotter.v1.windows\snapshots\32 is 18.7427368164063MB
Sum of C:\ProgramData\containerd\root\io.containerd.snapshotter.v1.windows\snapshots\33 is 0.107711791992188MB
Sum of C:\ProgramData\containerd\root\io.containerd.snapshotter.v1.windows\snapshots\34 is 532.011281967163MB
Sum of C:\ProgramData\containerd\root\io.containerd.snapshotter.v1.windows\snapshots\35 is 18.8944778442383MB
Sum of C:\ProgramData\containerd\root\io.containerd.snapshotter.v1.windows\snapshots\36 is 0.107711791992188MB
Sum of C:\ProgramData\containerd\root\io.containerd.snapshotter.v1.windows\snapshots\37 is 44.007116317749MB
Sum of C:\ProgramData\containerd\root\io.containerd.snapshotter.v1.windows\snapshots\38 is 63.7499618530273MB
Sum of C:\ProgramData\containerd\root\io.containerd.snapshotter.v1.windows\snapshots\39 is 66.4652786254883MB
Sum of C:\ProgramData\containerd\root\io.containerd.snapshotter.v1.windows\snapshots\4 is 5721.84680557251MB
Sum of C:\ProgramData\containerd\root\io.containerd.snapshotter.v1.windows\snapshots\40 is 83.8560028076172MB
Sum of C:\ProgramData\containerd\root\io.containerd.snapshotter.v1.windows\snapshots\41 is 46.4706649780273MB
Sum of C:\ProgramData\containerd\root\io.containerd.snapshotter.v1.windows\snapshots\42 is 46.9731979370117MB
Sum of C:\ProgramData\containerd\root\io.containerd.snapshotter.v1.windows\snapshots\43 is 58.3285598754883MB
Sum of C:\ProgramData\containerd\root\io.containerd.snapshotter.v1.windows\snapshots\44 is 58.6475143432617MB
Sum of C:\ProgramData\containerd\root\io.containerd.snapshotter.v1.windows\snapshots\45 is 58.3975143432617MB
Sum of C:\ProgramData\containerd\root\io.containerd.snapshotter.v1.windows\snapshots\46 is 58.5985946655273MB
Sum of C:\ProgramData\containerd\root\io.containerd.snapshotter.v1.windows\snapshots\47 is 531.96036529541MB
Sum of C:\ProgramData\containerd\root\io.containerd.snapshotter.v1.windows\snapshots\48 is 61.7398071289063MB
Sum of C:\ProgramData\containerd\root\io.containerd.snapshotter.v1.windows\snapshots\49 is 0.107711791992188MB
Sum of C:\ProgramData\containerd\root\io.containerd.snapshotter.v1.windows\snapshots\5 is 0.000344276428222656MB
Sum of C:\ProgramData\containerd\root\io.containerd.snapshotter.v1.windows\snapshots\50 is 532.781247138977MB
Sum of C:\ProgramData\containerd\root\io.containerd.snapshotter.v1.windows\snapshots\51 is 61.9639358520508MB
Sum of C:\ProgramData\containerd\root\io.containerd.snapshotter.v1.windows\snapshots\52 is 0.107711791992188MB
Sum of C:\ProgramData\containerd\root\io.containerd.snapshotter.v1.windows\snapshots\53 is 21.9458541870117MB
Sum of C:\ProgramData\containerd\root\io.containerd.snapshotter.v1.windows\snapshots\54 is 0.107711791992188MB
Sum of C:\ProgramData\containerd\root\io.containerd.snapshotter.v1.windows\snapshots\55 is 22.5893020629883MB
Sum of C:\ProgramData\containerd\root\io.containerd.snapshotter.v1.windows\snapshots\56 is 0.107711791992188MB
Sum of C:\ProgramData\containerd\root\io.containerd.snapshotter.v1.windows\snapshots\57 is 532.09653377533MB
Sum of C:\ProgramData\containerd\root\io.containerd.snapshotter.v1.windows\snapshots\58 is 0.107711791992188MB
Sum of C:\ProgramData\containerd\root\io.containerd.snapshotter.v1.windows\snapshots\59 is 52.7061080932617MB
Sum of C:\ProgramData\containerd\root\io.containerd.snapshotter.v1.windows\snapshots\6 is 0.000946044921875MB
Sum of C:\ProgramData\containerd\root\io.containerd.snapshotter.v1.windows\snapshots\60 is 0.107711791992188MB
Sum of C:\ProgramData\containerd\root\io.containerd.snapshotter.v1.windows\snapshots\61 is 53.4302215576172MB
Sum of C:\ProgramData\containerd\root\io.containerd.snapshotter.v1.windows\snapshots\62 is 0.107711791992188MB
Sum of C:\ProgramData\containerd\root\io.containerd.snapshotter.v1.windows\snapshots\63 is 54.9941024780273MB
Sum of C:\ProgramData\containerd\root\io.containerd.snapshotter.v1.windows\snapshots\64 is 0.107711791992188MB
Sum of C:\ProgramData\containerd\root\io.containerd.snapshotter.v1.windows\snapshots\65 is 67.1845169067383MB
Sum of C:\ProgramData\containerd\root\io.containerd.snapshotter.v1.windows\snapshots\66 is 0.107711791992188MB
Sum of C:\ProgramData\containerd\root\io.containerd.snapshotter.v1.windows\snapshots\67 is 67.2026977539063MB
Sum of C:\ProgramData\containerd\root\io.containerd.snapshotter.v1.windows\snapshots\68 is 0.107711791992188MB
Sum of C:\ProgramData\containerd\root\io.containerd.snapshotter.v1.windows\snapshots\69 is 68.7841262817383MB
Sum of C:\ProgramData\containerd\root\io.containerd.snapshotter.v1.windows\snapshots\7 is 0.0009765625MB
Sum of C:\ProgramData\containerd\root\io.containerd.snapshotter.v1.windows\snapshots\70 is 0.107711791992188MB
Sum of C:\ProgramData\containerd\root\io.containerd.snapshotter.v1.windows\snapshots\71 is 68.8158645629883MB
Sum of C:\ProgramData\containerd\root\io.containerd.snapshotter.v1.windows\snapshots\72 is 4030.40497303009MB
Sum of C:\ProgramData\containerd\root\io.containerd.snapshotter.v1.windows\snapshots\73 is 0.0390625MB
Sum of C:\ProgramData\containerd\root\io.containerd.snapshotter.v1.windows\snapshots\74 is 0.0390625MB
Sum of C:\ProgramData\containerd\root\io.containerd.snapshotter.v1.windows\snapshots\75 is 3.12396240234375MB
Sum of C:\ProgramData\containerd\root\io.containerd.snapshotter.v1.windows\snapshots\76 is 2.29451656341553MB
Sum of C:\ProgramData\containerd\root\io.containerd.snapshotter.v1.windows\snapshots\77 is 0.0390625MB
Sum of C:\ProgramData\containerd\root\io.containerd.snapshotter.v1.windows\snapshots\78 is 6.57713603973389MB
Sum of C:\ProgramData\containerd\root\io.containerd.snapshotter.v1.windows\snapshots\79 is 423.911822319031MB
Sum of C:\ProgramData\containerd\root\io.containerd.snapshotter.v1.windows\snapshots\8 is 60.68408203125MB
Sum of C:\ProgramData\containerd\root\io.containerd.snapshotter.v1.windows\snapshots\80 is 728.927686691284MB
Sum of C:\ProgramData\containerd\root\io.containerd.snapshotter.v1.windows\snapshots\81 is 7.54863262176514MB
Sum of C:\ProgramData\containerd\root\io.containerd.snapshotter.v1.windows\snapshots\82 is 0.0390625MB
Sum of C:\ProgramData\containerd\root\io.containerd.snapshotter.v1.windows\snapshots\83 is 0.0390625MB
Sum of C:\ProgramData\containerd\root\io.containerd.snapshotter.v1.windows\snapshots\84 is 0.0390625MB
Sum of C:\ProgramData\containerd\root\io.containerd.snapshotter.v1.windows\snapshots\85 is 0.0390625MB
Sum of C:\ProgramData\containerd\root\io.containerd.snapshotter.v1.windows\snapshots\86 is 0.0390625MB
Sum of C:\ProgramData\containerd\root\io.containerd.snapshotter.v1.windows\snapshots\87 is 0.0390625MB
Sum of C:\ProgramData\containerd\root\io.containerd.snapshotter.v1.windows\snapshots\88 is 32.4073991775513MB
Sum of C:\ProgramData\containerd\root\io.containerd.snapshotter.v1.windows\snapshots\89 is 0.0440912246704102MB
Sum of C:\ProgramData\containerd\root\io.containerd.snapshotter.v1.windows\snapshots\9 is 537.912572860718MB
Sum of C:\ProgramData\containerd\root\io.containerd.snapshotter.v1.windows\snapshots\90 is 571.441191673279MB
Sum of C:\ProgramData\containerd\root\io.containerd.snapshotter.v1.windows\snapshots\91 is 0.0883865356445313MB
Sum of C:\ProgramData\containerd\root\io.containerd.snapshotter.v1.windows\snapshots\92 is 0.0408115386962891MB
Sum of C:\ProgramData\containerd\root\io.containerd.snapshotter.v1.windows\snapshots\93 is 0.0963468551635742MB
Sum of C:\ProgramData\containerd\root\io.containerd.snapshotter.v1.windows\snapshots\94 is 0.0409488677978516MB
Sum of C:\ProgramData\containerd\root\io.containerd.snapshotter.v1.windows\snapshots\95 is 0.172444343566895MB
Sum of C:\ProgramData\containerd\root\io.containerd.snapshotter.v1.windows\snapshots\96 is 88.122974395752MB
Sum of C:\ProgramData\containerd\root\io.containerd.snapshotter.v1.windows\snapshots\97 is 17.9332962036133MB
Sum of C:\ProgramData\containerd\root\io.containerd.snapshotter.v1.windows\snapshots\98 is 18.1525287628174MB
Sum of C:\ProgramData\containerd\root\io.containerd.snapshotter.v1.windows\snapshots\99 is 0.0398521423339844MB
Total size of all snapshots: 34429.4702396393MB

AbelHu reviewed (approved):
LGTM. Thanks for the PR.
